### PR TITLE
feat: Strip surrogates for postgres

### PR DESCRIPTION
**Pull request by mitsuhiko**
This removes lone surrogates on the way to postgres.  This works around issues
that manifest like this: "invalid byte sequence for encoding "UTF8": 0xed 0xa0 0xbd".

It's similar to how we deal with null values now.

### CODE REVIEW

**Comment by mitsuhiko** [comment]
Going to add a test, so don't merge yet.

**Review by jan-auer — APPROVED** [review]
🤷‍♀️

**Comment by mitsuhiko** [comment]
Would like a review from matt or ops before I merge this just to make sure it does not cause any issues. Should resolve around 37 sentry errors and improve our event ingestion rate for javascript.

File: src/sentry/utils/strings.py
Comment (markstory) [inline src/sentry/utils/strings.py]: :+1: Ranges match high/low surrogate pair ranges [here](https://en.wikipedia.org/wiki/UTF-16)

**Pull request merged**
